rust/src/enip/enip.rs

```diff
@@ -0,0 +1,480 @@
+                }
+                Err(nom::Err::Incomplete(_)) => {
+                    let consumed = input.len() - start.len();
+                    let needed = start.len() + 1;
+                    return AppLayerResult::incomplete(consumed as u32, needed as u32);
+                }
+                Err(_) => {
+                    return AppLayerResult::err();
+                }
+            }
+        }
+
+        // All input was fully consumed.
+        return AppLayerResult::ok();
+    }
+
+    fn on_request_gap(&mut self, _size: u32) {
+        self.request_gap = true;
+    }
+
+    fn on_response_gap(&mut self, _size: u32) {
+        self.response_gap = true;
+    }
+}
+
+/// Probe for a valid header.
+///
+/// As this enip protocol uses messages prefixed with the size
+/// as a string followed by a ':', we look at up to the first 10
+/// characters for that pattern.
+fn probe(input: &[u8]) -> bool {
+    match parser::parse_enip_header(input) {
+        Ok((rem, header)) => {
+            match header.status {
+                parser::ENIP_STATUS_SUCCESS
+                | parser::ENIP_STATUS_INVALID_CMD
+                | parser::ENIP_STATUS_NO_RESOURCES
+                | parser::ENIP_STATUS_INCORRECT_DATA
+                | parser::ENIP_STATUS_INVALID_SESSION
+                | parser::ENIP_STATUS_INVALID_LENGTH
+                | parser::ENIP_STATUS_UNSUPPORTED_PROT_REV
+                | parser::ENIP_STATUS_ENCAP_HEADER_ERROR => {} // Ok so far, continue
+                _ => {
+                    return false;
+                }
+            }
+
+            match header.cmd {
+                parser::ENIP_CMD_NOP => {
+                    if header.options != 0 {
+                        return false;
+                    }
+                }
+                parser::ENIP_CMD_REGISTER_SESSION => {
+                    if header.pdulen != 4 {
+                        return false;
+                    }
+                }
+                parser::ENIP_CMD_UNREGISTER_SESSION => {
+                    if header.pdulen != 4 && header.pdulen != 0 {
+                        return false;
+                    }
+                }
+                parser::ENIP_CMD_LIST_INTERFACES => {
+                    if !parser::parse_enip_list_interfaces(rem).is_ok() {
+                        return false;
+                    }
+                }
+                parser::ENIP_CMD_LIST_SERVICES
+                | parser::ENIP_CMD_LIST_IDENTITY
+                | parser::ENIP_CMD_SEND_RRDATA
+                | parser::ENIP_CMD_SEND_UNIT_DATA
+                | parser::ENIP_CMD_INDICATE_STATUS
+                | parser::ENIP_CMD_CANCEL => {} // Ok so far, continue
+                _ => {
+                    return false;
+                }
+            }
+            return true;
+        }
+        _ => {
+            return false;
+        }
+    }
+}
+
+// C exports.
+
+unsafe extern "C" fn rs_enip_probing_parser_udp(
```

Comment:
Non-public, we can drop the `rs` prefix, just basic Rust naming.  Yeah, this is a break from the previous convention, which doesn't really make much sense anymore IMO.

Reply:
Ok for all cf `unsafe extern "C" fn  rs`

Reply:
> Ok for all cf `unsafe extern "C" fn rs`

I think so.  Cause they don't become public linkable so don't form any part of the public C API.